### PR TITLE
CMake: add support for BOUND_CHECK

### PR DESCRIPTION
**Pull request by mic84**
## Checklist

The proposed changes:
- [ ] fix a bug or incorrect behavior in AMReX
- [ ] add new capabilities to AMReX
- [ ] changes answers in the test suite to more than roundoff level
- [ ] are likely to significantly affect the results of downstream AMReX users
- [ ] are described in the proposed changes to the AMReX documentation, if appropriate

### CODE REVIEW

**Review by WeiqunZhang — APPROVED** [review]

File: Tools/CMake/AMReXOptions.cmake [hunk 1]
```diff
@@ -283,6 +283,8 @@ endif ()
 
 print_option( AMReX_ASSERTIONS )
 
+option(AMReX_BOUND_CHECK  "Enable bound checking in Array4 class" OFF)
```
Comment (ax3l) [inline Tools/CMake/AMReXOptions.cmake:286]: Oh cool, should we enable this by default with Debug builds?
How is the corresponding GNUmake logic?

Comment (WeiqunZhang) [reply Tools/CMake/AMReXOptions.cmake:286]: It's enabled with debug builds.

Comment (WeiqunZhang) [reply Tools/CMake/AMReXOptions.cmake:286]: My message was probably not very clear.  The `AMREX_BOUND_CHECK` macro is only used like below
```
#if defined(AMREX_DEBUG) || defined(AMREX_BOUND_CHECK)
```

Comment (ax3l) [reply Tools/CMake/AMReXOptions.cmake:286]: Ok perfect, then we can keep the build system logic simple in both GNUmake & CMake.

**Pull request merged**